CMakeLists.txt

```diff
@@ -11,64 +11,70 @@ if(WITH_CUDA)
-# Image extension
-file(GLOB IMAGE_HEADERS torchvision/csrc/cpu/image/*.h)
-file(GLOB IMAGE_SOURCES torchvision/csrc/cpu/image/*.cpp)
-file(GLOB OPERATOR_SOURCES torchvision/csrc/cpu/*.h torchvision/csrc/cpu/*.cpp ${IMAGE_HEADERS} ${IMAGE_SOURCES} ${HEADERS} torchvision/csrc/*.cpp)
 if(WITH_CUDA)
-  file(GLOB OPERATOR_SOURCES ${OPERATOR_SOURCES} torchvision/csrc/cuda/*.h torchvision/csrc/cuda/*.cu)
+  # Gather cuda source files.
+  file(GLOB OPERATOR_CUDA_SOURCES torchvision/csrc/cuda/*.cu)
+  list(APPEND OPERATOR_SOURCES ${OPERATOR_CUDA_SOURCES})
+
+  # Set cuda include directories.
+  include_directories(torchvision/cinclude/${PROJECT_NAME}_cuda)
 endif()
-file(GLOB MODELS_HEADERS torchvision/csrc/models/*.h)
-file(GLOB MODELS_SOURCES torchvision/csrc/models/*.h torchvision/csrc/models/*.cpp)
 
-add_library(${PROJECT_NAME} SHARED ${MODELS_SOURCES} ${OPERATOR_SOURCES} ${IMAGE_SOURCES})
-target_link_libraries(${PROJECT_NAME} PRIVATE ${TORCH_LIBRARIES} ${PNG_LIBRARY} Python3::Python)
-# target_link_libraries(${PROJECT_NAME} PRIVATE ${PNG_LIBRARY} Python3::Python)
+# Add library.
+add_library(${PROJECT_NAME} SHARED ${MODELS_SOURCES} ${OPERATOR_SOURCES})
+target_link_libraries(${PROJECT_NAME} PRIVATE ${TORCH_LIBRARIES} Python3::Python)
 set_target_properties(${PROJECT_NAME} PROPERTIES EXPORT_NAME TorchVision)
 
-target_include_directories(${PROJECT_NAME} INTERFACE
-  $<BUILD_INTERFACE:${HEADERS}:${PNG_INCLUDE_DIR}>
-  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
-
 include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
 
 set(TORCHVISION_CMAKECONFIG_INSTALL_DIR "share/cmake/TorchVision" CACHE STRING "install path for TorchVisionConfig.cmake")
 
 configure_package_config_file(cmake/TorchVisionConfig.cmake.in
-  "${CMAKE_CURRENT_BINARY_DIR}/TorchVisionConfig.cmake"
-  INSTALL_DESTINATION ${TORCHVISION_CMAKECONFIG_INSTALL_DIR})
+    "${CMAKE_CURRENT_BINARY_DIR}/TorchVisionConfig.cmake"
+    INSTALL_DESTINATION ${TORCHVISION_CMAKECONFIG_INSTALL_DIR}
+)
 
 write_basic_package_version_file(${CMAKE_CURRENT_BINARY_DIR}/TorchVisionConfigVersion.cmake
-  VERSION ${TORCHVISION_VERSION}
-  COMPATIBILITY AnyNewerVersion)
+    VERSION ${TORCHVISION_VERSION}
+    COMPATIBILITY AnyNewerVersion
+)
 
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/TorchVisionConfig.cmake
-  ${CMAKE_CURRENT_BINARY_DIR}/TorchVisionConfigVersion.cmake
-  DESTINATION ${TORCHVISION_CMAKECONFIG_INSTALL_DIR})
+    ${CMAKE_CURRENT_BINARY_DIR}/TorchVisionConfigVersion.cmake
+    DESTINATION ${TORCHVISION_CMAKECONFIG_INSTALL_DIR}
+)
 
+# Install library.
 install(TARGETS ${PROJECT_NAME}
-  EXPORT TorchVisionTargets
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-  )
+    EXPORT TorchVisionTargets
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
 
 install(EXPORT TorchVisionTargets
-  NAMESPACE TorchVision::
-  DESTINATION ${TORCHVISION_CMAKECONFIG_INSTALL_DIR})
+    NAMESPACE TorchVision::
+    DESTINATION ${TORCHVISION_CMAKECONFIG_INSTALL_DIR}
+)
```

Comment:
These all look like unrelated formatting changes. I'd say stick to the current style, and if you want to propose a different one do so in a different PR.